module/product/src/Infrastructure/Mapper/Strategy/ProductRelationAttributeMapperStrategy.php

```diff
@@ -8,37 +8,56 @@
 
 namespace Ergonode\Product\Infrastructure\Mapper\Strategy;
 
+use Ergonode\Attribute\Infrastructure\Mapper\Strategy\ContextAwareAttributeMapperStrategyInterface;
+use Ergonode\SharedKernel\Domain\AggregateId;
 use Ergonode\Value\Domain\ValueObject\ValueInterface;
 use Ergonode\Attribute\Domain\ValueObject\AttributeType;
 use Webmozart\Assert\Assert;
 use Ergonode\Value\Domain\ValueObject\StringCollectionValue;
 use Ergonode\Product\Domain\Entity\Attribute\ProductRelationAttribute;
-use Ergonode\Attribute\Infrastructure\Mapper\Strategy\AttributeMapperStrategyInterface;
 
-class ProductRelationAttributeMapperStrategy implements AttributeMapperStrategyInterface
+class ProductRelationAttributeMapperStrategy implements ContextAwareAttributeMapperStrategyInterface
 {
     public function supported(AttributeType $type): bool
     {
         return $type->getValue() === ProductRelationAttribute::TYPE;
     }
 
-    public function map(array $values): ValueInterface
+    public function map(array $values, ?AggregateId $aggregateId = null): ValueInterface
     {
         Assert::allRegex(array_keys($values), '/^[a-z]{2}_[A-Z]{2}$/');
-
+        Assert::notNull($aggregateId);
+        $mappedValues = [];
         foreach ($values as $language => $value) {
             if (null !== $value && !is_array($value)) {
                 $value = explode(',', (string) $value);
             }
 
             if (is_array($value) && !empty($value)) {
                 Assert::allUuid($value);
-                $values[$language] = implode(',', $value);
+                $value = $this->filter($value, $aggregateId);
+
+                if (empty($value)) {
+                    continue;
+                }
+                $mappedValues[$language] = implode(',', $value);
             } else {
-                $values[$language] = null;
+                $mappedValues[$language] = null;
+            }
+        }
+
+        return new StringCollectionValue($mappedValues);
+    }
+
+    private function filter(array $value, AggregateId $aggregateId): ?array
+    {
+        $result = [];
+        foreach ($value as $item) {
+            if ($item !== $aggregateId->getValue()) {
+                $result[] = $item;
             }
         }
 
-        return new StringCollectionValue($values);
+        return !empty($result) ? $result : null;
```

Comment:
```suggestion
        return empty($result) ? null : $result;
```
Try not to use negations in such simple cases. They are just harder to get